lib/components/map/connected-transit-vehicle-overlay.js

```diff
@@ -102,15 +102,28 @@ function VehicleTooltip(props) {
   )
 }
 
-// Round vehicle symbol with arrow/caret on the border,
-// and showing the route color with a transparent effect on hover.
-const IconContainer = withRouteColorBackground(
-  withCaret(Circle, { height: 5, offset: 1.5, width: 10 }),
-  {
+/**
+ *
+ * @param {Object} props vehicle object (includes heading)
+ * @returns circular mode icon overlay that includes a caret only if bearing is not null
+ */
+const IconContainer = (props) => {
+  const { vehicle } = props
+  const caretHeight = vehicle?.heading === null ? -1 : 5
+
+  const OverlayIcon = withCaret(Circle, {
```

Comment:
This sounds like a patch is needed in the transit vehicle overlay package instead. For performance reasons, we don't want to be generating component classes in the rendering code of another component.

Reply:
Yeah agreed the function is clunky. Do you mean changing the TransitVehicleOverlay function directly? Right now it looks like it's in my node_modules from @opentripplnner, would this need to be patched in another repo?

Reply:
Update: I will take a stab at patching the `TransitVehicleOverlay` in `otp-ui`
PR will be linked here after it is made